released/discovery_center/mission_3252/main.tf

```diff
@@ -0,0 +1,80 @@
+###############################################################################################
+# Setup of names in accordance to naming convention
+###############################################################################################
+resource "random_uuid" "uuid" {}
+
+locals {
+  random_uuid               = random_uuid.uuid.result
+  project_subaccount_domain = lower(replace("mission-3252-${local.random_uuid}", "_", "-"))
+}
+
+###############################################################################################
+# Creation of subaccount
+###############################################################################################
+resource "btp_subaccount" "project" {
+  name      = var.subaccount_name
+  subdomain = local.project_subaccount_domain
+  region    = lower(var.region)
+}
+
+###############################################################################################
+# Assignment of users as sub account administrators
+###############################################################################################
+resource "btp_subaccount_role_collection_assignment" "subaccount-admins" {
+  for_each             = toset("${var.subaccount_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Administrator"
+  user_name            = each.value
+}
+
+###############################################################################################
+# Assignment of users as sub account service administrators
+###############################################################################################
+resource "btp_subaccount_role_collection_assignment" "subaccount-service-admins" {
+  for_each             = toset("${var.subaccount_service_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Service Administrator"
+  user_name            = each.value
+}
+
+######################################################################
+# Setup Kyma
+######################################################################
+data "btp_regions" "all" {}
+
+data "btp_subaccount" "this" {
+  id = btp_subaccount.project.id
+}
+
+locals {
+  subaccount_iaas_provider = [for region in data.btp_regions.all.values : region if region.region == data.btp_subaccount.this.region][0].iaas_provider
+}
+
+resource "btp_subaccount_entitlement" "kymaruntime" {
+  subaccount_id = btp_subaccount.project.id
+  service_name  = "kymaruntime"
+  plan_name     = lower(local.subaccount_iaas_provider)
```

Comment:
Shouldn't the plan name be a variable that the user can choose?

If this value should be determined dynamically then the logic should be explained via a comment in line 49. I also doubt that this is correct as the data source for subaccount has field called [region](https://registry.terraform.io/providers/SAP/btp/latest/docs/data-sources/subaccount#region), but this is a string and not an array.